tests/installation/add_maintenance_repos.pm

```diff
@@ -0,0 +1,31 @@
+# SUSE's openQA tests
+#
+# Copyright 2022 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: Add new add-ons using maintenance test repo URLs.
+#
+# Maintainer: QA SLE YaST team <qa-sle-yast@suse.de>
+
+use base 'y2_installbase';
+use strict;
+use warnings;
+use testapi 'get_var';
+
+sub run {
+    set_var('MAINT_TEST_REPO', get_var('INCIDENT_REPO')) if get_var('INCIDENT_REPO');
+    my @repos = split(/,/, get_var('MAINT_TEST_REPO'));
+
+    $testapi::distri->get_add_on_product()->confirm_like_additional_add_on();
+
+    while (defined(my $maintrepo = shift @repos)) {
+        $testapi::distri->get_add_on_product()->accept_current_media_type_selection();
+        $testapi::distri->get_repository_url()->add_repo({url => $maintrepo});
+        if (@repos) {
```

Comment:
this is not needed, just iterate the array please.
btw, no need for screenshots here, it would be an overkill.

Reply:
It is needed, because in the case this is the last item (meaning `@repos` now returns false due to last shift) the code must not press the button to add a new repo, but instead press next and continue with the installation.

This cannot take place at the begining of the loop (as it must not take place during the first iteration), so it needs to be conditional.

There are other ways to do it (like having the first iteration outside of the loop) but it leads to lengthier and even more confusing code. On the other hand, this is also a replication of the current behavior of the (non-libyui) module.

Reply:
the code should reflect what is happening which is ... (no surprise, a bad UI design, but it is what it is). Therefore I would expect to see:
- Add-on-product page: click checkbox 'I would like to ...'
- (optional) I think there is not need to click on "specify url' radiobutton as it is  selected by default.
- Repository-url page: enter name and url and next
- Add-on-product-installation-page: click on add button
- (optional) add-on-product page: looks like the radiobutton is still selected by default and now we don't have "I would like to ...". Therefore this step should be "accept current options method" so basically just next behind the scene.

So yes, it would be more helpful to shift the first element and treat it differently as the UI is also different and then do the loops with N-1. It exactly how UI does. It would mimic better the behavior we see.

Reply:
Agreed, that kind of conditional problem (first iteration outside vs conditional action inside the loop) is always annoying. I almost always chose the `less code` option by default, even if it means one extra conditional for every loop iteration, but that's just habit.

I re-wrote it removing the loop and using foreach, running a ver run and update the pr: https://openqa.suse.de/tests/9386975#live

Reply:
Done